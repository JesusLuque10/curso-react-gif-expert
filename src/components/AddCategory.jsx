import { useState } from 'react';
import Proptypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const OnInputChange = ({ target }) => {
    //el event esta desestructurado y sacamos el target
    setInputValue(target.value);
  };

  const OnSubmit = (event) => {
    event.preventDefault(); //para que no recargue
    if (inputValue.trim().length <= 1) return;

    onNewCategory(inputValue);
    setInputValue('');
  };

  return (
    <form onSubmit={OnSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        //onChange={(event) => OnInputChange(event)}
        onChange={OnInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: Proptypes.func.isRequired,
};
