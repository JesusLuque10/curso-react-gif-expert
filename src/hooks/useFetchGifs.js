import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/GetGifByCategory';

export const useFetchGifs = (category) => {
  const [images, setIamges] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  const getImages = async () => {
    const newImages = await getGifs(category);
    setIamges(newImages);
    setIsLoading(false);
  };

  useEffect(() => {
    getImages();
  }, []); //si esta vacio significa que solo se va a ejecutar la primera vez que se construye el componenete

  return {
    images,
    isLoading,
  };
};
