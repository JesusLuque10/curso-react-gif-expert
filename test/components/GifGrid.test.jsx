import { render, screen } from '@testing-library/react';
import { GifGrid } from '../../src/components/GifGrid';
import { useFetchGifs } from '../../src/hooks/useFetchGifs';

jest.mock('../../src/hooks/useFetchGifs');

describe('Pruebas del componente <GifGrid/>', () => {
  var category = 'One Punch';

  test('Debe mostrar el loading', () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true,
    });

    render(<GifGrid category={category} />);

    //screen.debug();
    expect(screen.getByText('...Cargando'));
    expect(screen.getByText(category));
  });

  test('Deben cargarse las imagenes al hacer el useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        title: 'Saitama',
        url: 'http:ofgofign',
      },
      {
        id: '54654',
        title: 'Pokemon',
        url: 'http:ofgofign',
      },
    ];

    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false,
    });

    render(<GifGrid category={category} />);
    //screen.debug();
    expect(screen.getAllByRole('img').length).toBe(2);
  });

  test('Debe hacer match con el Snapshot', () => {
    const gifs = [
      {
        id: 'ABC',
        title: 'Saitama',
        url: 'http:ofgofign',
      },
      {
        id: '54654',
        title: 'Pokemon',
        url: 'http:ofgofign',
      },
    ];

    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false,
    });

    const container = render(<GifGrid category={category} />);
    expect(container).toMatchSnapshot();
  });
});
