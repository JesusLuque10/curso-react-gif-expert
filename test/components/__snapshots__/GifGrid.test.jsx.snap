// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGrid/> Debe hacer match con el Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3>
        One Punch
      </h3>
      <div
        class="card-grid"
      >
        <div
          class="card"
        >
          <img
            alt="Saitama"
            src="http:ofgofign"
          />
          <p>
            Saitama
          </p>
        </div>
        <div
          class="card"
        >
          <img
            alt="Pokemon"
            src="http:ofgofign"
          />
          <p>
            Pokemon
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h3>
      One Punch
    </h3>
    <div
      class="card-grid"
    >
      <div
        class="card"
      >
        <img
          alt="Saitama"
          src="http:ofgofign"
        />
        <p>
          Saitama
        </p>
      </div>
      <div
        class="card"
      >
        <img
          alt="Pokemon"
          src="http:ofgofign"
        />
        <p>
          Pokemon
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pruebas del componente <GifGrid/> Deben cargarse las imagenes al hacer el useFetchGifs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3>
        One Punch
      </h3>
      <div
        class="card-grid"
      >
        <div
          class="card"
        >
          <img
            alt="Saitama"
            src="http:ofgofign"
          />
          <p>
            Saitama
          </p>
        </div>
        <div
          class="card"
        >
          <img
            alt="Pokemon"
            src="http:ofgofign"
          />
          <p>
            Pokemon
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h3>
      One Punch
    </h3>
    <div
      class="card-grid"
    >
      <div
        class="card"
      >
        <img
          alt="Saitama"
          src="http:ofgofign"
        />
        <p>
          Saitama
        </p>
      </div>
      <div
        class="card"
      >
        <img
          alt="Pokemon"
          src="http:ofgofign"
        />
        <p>
          Pokemon
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
